libc-top-half/musl/arch/wasm32/pthread_arch.h

```diff
@@ -1,12 +1,11 @@
-#ifdef _REENTRANT
-#error "multiple threads not supported in musl yet"
+uintptr_t __get_tp(void) {
+#if _REENTRANT
+  int val;
+  __asm__(".globaltype g_needs_dynamic_alloc, i32\n"
+          "global.get __wasi_libc_pthread_self\n"
+          "local.set %0" : "=r" (val));
+  return val;
+#else
+  return 0;
```

Comment:
Does `__get_tp` need to be defined in non-`_REENTRANT` mode? It seems better to get an undefined-symbol link error if we try to use it than to have it silently return NULL, especially with NULL being a valid address.

Reply:
Thanks for all the comments.  I will address all of them, this PR is certainly not ready to land.  I just wanted to share what I had done so far.

Reply:
Yep, makes sense. I was just looking through it so that I could follow along, and happened to have comments along the way :-).